File: EJERCICIOS/7.py
"""
Una persona que va de compras a la multitienda "CyberSTORE" decide llevar 
un control sobre lo que va comprando, para saber la cantidad de dinero 
que tendrá que pagar al llegar a la caja
La tienda solamente vende 3 tipos de productos, estos pueden ser 
Televisores (codigo 1), Refrigeradores (codigo 2), Lavadoras (codigo 3)
No se puede ingresar el codigo de otro articulo, por lo que solo esos 3
valores pueden ser ingresados (validacion)
La persona ira exactamente 3 veces a la tienda...

Ud. debe realizar un Algoritmo que permita, en cada vez que visite la tienda se:
    > Indicar el codigo
    > Ingresar la cantidad
    > Ingresar el valor

Finalmente, el algoritmo debe dar como respuesta lo siguiente:

Se compraron X televisores, a un total de XXXXX
Se compraron Y refrigeradores, a un total de XXXXX
Se compraron Z lavadoras, a un total de XXXXX
"""

def validar(x):
    if(x==1): return True
    elif(x==2): return True
    elif(x==3): return True
    else: return False

def ingresarDatos():
    print('Seleccione su producto')
    print('Televisores (codigo 1), Refrigeradores (codigo 2), Lavadoras (codigo 3)')
    while(range(3)<3):
        print(f"Ingresar codigo")
        codigo = int(input())
        if(validar(codigo)):
            print(f"Ingresar la cantidad")
            cantidad = int(input())
            print(f"Ingresar el valor")
            valor = float(input())
        else: ingresarDatos(); break; 

    asignaturas=("Programacion","Algebra Lineal","Ing de Sistemas")
    datos = []
    for asignatura in asignaturas:
        print(f"Ingrese la nota de examen de {asignatura}")
        examen = float(input())
        print(f"Ingrese la nota de practica 1  de {asignatura}")
        practica1 = float(input())
        print(f"Ingrese la nota de practica 2 de {asignatura}")
        practica2 = float(input())
        datos.append({"asignatura":asignatura,"examen":examen,
                      "practica1":practica1,"practica2":practica2})
    return datos

File: EJERCICIOS/typescript/7.ts
import readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

const rl = readline.createInterface({ input, output });

type Producto = {
  codigo: number;
  cantidad: number;
  valor: number;
};

function validar(codigo: number): boolean {
  return codigo === 1 || codigo === 2 || codigo === 3;
}

async function ingresarDatos(): Promise<Producto[]> {
  console.log('Seleccione su producto...');
  console.log('Televisores (codigo 1), Refrigeradores (codigo 2), Lavadoras (codigo 3)');

  const productos: Producto[] = [];

  for (let i = 0; i < 3; i++) {
    console.log('Ingresar codigo');
    const codigo = Number(await rl.question(''));

    if (validar(codigo)) {
      console.log('Ingresar la cantidad');
      const cantidad = Number(await rl.question(''));

      console.log('Ingresar el valor');
      const valor = Number(await rl.question(''));

      productos.push({ codigo, cantidad, valor });
    } else {
      console.log('error');
    }
  }

  return productos;
}

async function calculo1(): Promise<void> {
  const productos = await ingresarDatos();
  let cantidad1 = 0;
  let cantidad2 = 0;
  let cantidad3 = 0;
  let valor1 = 0;
  let valor2 = 0;
  let valor3 = 0;

  for (const producto of productos) {
    if (producto.codigo === 1) {
      cantidad1 += producto.cantidad;
      valor1 += producto.valor * producto.cantidad;
    }

    if (producto.codigo === 2) {
      cantidad2 += producto.cantidad;
      valor2 += producto.valor * producto.cantidad;
    }

    if (producto.codigo === 3) {
      cantidad3 += producto.cantidad;
      valor3 += producto.valor * producto.cantidad;
    }
  }

  console.log(`Se compraron ${cantidad1} televisores, a un total de ${valor1}`);
  console.log(`Se compraron ${cantidad2} refrigeradores, a un total de ${valor2}`);
  console.log(`Se compraron ${cantidad3} lavadoras, a un total de ${valor3}`);
}

await calculo1();
rl.close();

